### PR TITLE
feat(issue-states): Add presave checks for GroupSubStatus

### PR DESCRIPTION
**Pull request by snigdhas**
* Add presave checks for GroupSubStatus to ensure we don't end up with inconsistent status/substatus pairings
* Add substatuses in the code when we change the status values
* Update tests to include substatuses in test data
* Add tests to check the substatus presave validation

WOR-2973

### CODE REVIEW

File: src/sentry/models/group.py [hunk 1]
```diff
@@ -718,5 +736,26 @@ def issue_category(self):
...
 def pre_save_group_default_substatus(instance, sender, *args, **kwargs):
     if instance:
-        if instance.status == GroupStatus.UNRESOLVED and instance.substatus is None:
-            instance.substatus = GroupSubStatus.ONGOING
+        # We only support substatuses for UNRESOLVED and IGNORED groups
+        if (
+            instance.status not in [GroupStatus.UNRESOLVED, GroupStatus.IGNORED]
+            and instance.substatus is not None
+        ):
+            raise ValueError(
```
Comment (armenzg) [inline src/sentry/models/group.py:744]: I _wonder_ if we should use `logger.exception` first and then on a follow up PR make all the ValueError changes.

What do you think?

Maybe I'm paranoid.

**Review by armenzg — COMMENTED** [review]
Thanks for all the effort on this PR! It must have not been the most fun to go through all those files.

File: src/sentry/models/group.py (hunk 1 above)
Comment (snigdhas) [reply src/sentry/models/group.py]: that's a great idea - we can let it run for a week and switch to ValueError if things look safe

**Review by armenzg — APPROVED** [review]
🎉

**Pull request merged**